src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -475,6 +475,12 @@ protected virtual Task<bool> HandleSignOutCallbackAsync()
                 authorizationResponse = messageReceivedContext.ProtocolMessage;
                 properties = messageReceivedContext.Properties;
 
+                // if any of the error fields are set, throw error null
+                if (!string.IsNullOrEmpty(authorizationResponse.Error))
```

Comment:
IMHO, that would be better to fix the OAuth2 handler to validate the `state`/XSRF cookie before handling the error.